include/envoy/network/udp_packet_writer_handler.h

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+
+#include "envoy/api/io_error.h"    // IoCallUint64Result
+#include "envoy/buffer/buffer.h"   // Buffer
+#include "envoy/network/address.h" // Address
+#include "envoy/network/socket.h"  // Socket
+#include "envoy/stats/scope.h"
+#include "envoy/stats/stats_macros.h"
+
+namespace Envoy {
+namespace Network {
+
+/**
+ * Max v6 packet size, excluding IP and UDP headers.
+ */
+static const uint64_t K_MAX_OUTGOING_PACKET_SIZE = 1452;
+
+/**
+ * UdpPacketWriterBuffer bundles a buffer and a function that
+ * releases it.
+ */
+struct UdpPacketWriterBuffer {
```

Comment:
This is a bit of a scary interface as there is no length. Can you add a length here so that we can at least do bounds checking where appropriate? Also, can we reuse any of the existing buffer interface like `Slice`? cc @antoniovicente. Finally, prefer `uint8_t` vs. `char` for this type of thing.

Reply:
The length is not provided from quiche interface actually. It guarantee that the buffer is no smaller than 1452. This is based on ethernet's max size, minus the IP and UDP headers. IPv6 has a 40 byte header, UDP adds an additional 8 bytes.  This is a total overhead of 48 bytes.  Ethernet's max packet size is 1500 bytes,  1500 - 48 = 1452.  

The writer impl is responsible to manage the memory by calling releaser_buffer_ at the right time.

The user of the writer only need to make sure not to write more than 1452 bytes.

We can probably use fragment with fixed value 1452 for len_.

Reply:
I have added the `length` filed to `UdpPacketWriterBuffer`. 
As Dan suggested, I am currently setting the length as 1452 bytes via `Network::K_MAX_OUTGOING_PACKET_SIZE`.

Also, I am using `uint8_t` instead of `char` now!

PS: I will try to replace `UdpPacketWritterBuffer` with `Buffer::OwnedBufferFragmentImpl` in the following changes!

Reply:
It sounds like you're requiring this buffer to be of some specific size that is not in any way related to the maximum UDP packet size.  It would be good to pass in length explicitly.